src/js/angular/clustermanagement/services/cluster-drawing.service.js

```diff
@@ -307,6 +324,12 @@ function resizeLabelDynamic(nodes) {
             const displayMessage = isShorten ? message : d.recoveryStatus.message;
             const width = calculateElementSizeByText(displayMessage).width;
             const height = calculateElementSizeByText(displayMessage).height;
+            object.on('.tooltip', null);
+            if (isShorten) {
+                addRecoveryLabelListeners(object, d.recoveryStatus.message, width, height, displayMessage);
```

Comment:
The way D3 works with datasets is that you define `enter/exit` functions which are applied to action on entries in the dataset.

When you add node, or when you remove a node.

This in mind I think you should unsubscribe in the `exit` phase, because the dataset could have changed.

@teodossidossev WDYT?

Reply:
Absolutely. If you don't unsubscribe on exit, another listener will be added to the object every time you enter it again. This accumulation will lead to unpredicted behavior.

Reply:
The documentation on event handling in `D3` (https://d3js.org/d3-selection/events) doesn't give any recommendation about where the listeners should be removed. Trying to remove them in the `exit` phase caused other problems, where node removal did not happen as it should. 
After testing the current code, it appears that these listeners are removed correctly.